include/vulkan/utility/vk_format_utils.h

```diff
@@ -0,0 +1,2297 @@
+// *** THIS FILE IS GENERATED - DO NOT EDIT ***
+// See format_utils_generator.py for modifications
+// Copyright 2023 The Khronos Group Inc.
+// Copyright 2023 Valve Corporation
+// Copyright 2023 LunarG, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <vulkan/vulkan.h>
+
+#include <stdbool.h>
+
+#define FORMAT_UTILS_MAX_PLANES 3
+#define FORMAT_UTILS_MAX_COMPONENTS 4
+
+enum FORMAT_UTILS_NUMERICAL_TYPE {
+    FORMAT_UTILS_NUMERICAL_TYPE_NONE = 0,
+    FORMAT_UTILS_NUMERICAL_TYPE_SFLOAT,
+    FORMAT_UTILS_NUMERICAL_TYPE_SINT,
+    FORMAT_UTILS_NUMERICAL_TYPE_SNORM,
+    FORMAT_UTILS_NUMERICAL_TYPE_SRGB,
+    FORMAT_UTILS_NUMERICAL_TYPE_SSCALED,
+    FORMAT_UTILS_NUMERICAL_TYPE_UFLOAT,
+    FORMAT_UTILS_NUMERICAL_TYPE_UINT,
+    FORMAT_UTILS_NUMERICAL_TYPE_UNORM,
+    FORMAT_UTILS_NUMERICAL_TYPE_USCALED
+};
+
+enum FORMAT_UTILS_COMPATIBILITY_CLASS {
```

Comment:
1. would it be better to forward declare so people looking at the file can quickly see the functions at the top?
2. I guess curious if this should be `VK_FORMAT_UTILS_*`... should we use `VK_FORMAT_UTILS` name space to match the filename

Reply:
1. Forward declaring the enums is a good idea, I made the functions all forward declared for the same logic after all. 

2. VK_ looks like a vulkan api enum which I do not want to be mistaken as.

Reply:
>  1. would it be better to forward declare so people looking at the file can quickly see the functions at the top?

I personally disagree on forward declaring. Most IDEs offer a way to collapse all functions/scopes. So IMO the forward declarations just adds more clutter/complexity.

Reply:
> VK_ looks like a vulkan api enum which I do not want to be mistaken as.

FFTI. I think using `VUL_` would be more appropriate.

Reply:
Forward declaring can be annoying since `enum` doesn't declare a size, but sizes are needed for forward declarations. I am not going to try to do that here, since I don't want to delay this PR any further while I wrestle getting enum forward decls working in C++ & C.

Reply:
Agreed that's why I said FFTI.